### PR TITLE
Improve README for Pint CLI repo on github (CSE-298)

### PR DESCRIPTION
**Pull request by chajain**

### CODE REVIEW

**Review by guangyee — APPROVED** [review]
lgtm

**Review by jeremy-moffitt — COMMENTED** [review]
So if I wanted to test local changes would I "install from source" using `pip install -e .[dev]` every time or is there another workflow?

File: README.md
Comment (rtamalin) [inline README.md:16]: I would recommend creating a virtrualenv in which to do the pip installation; installing a Python module via pip directly into the system Python infrastructure can corrupt that infrastructure, and in the case of SLE, could invalidate the customer's support contract. ;-)

File: README.md
Comment (rtamalin) [inline README.md:28]: I would recommend moving this up to follow the pip install instructions, since the `-e` makes it a developer mode install of the package, i.e. creates symlinks back to the cloned git repo directory... Again this really should be installing under a virtualenv, and not directly into the system Python infrastructure, especially for a developer mode install...

File: README.md
Comment (rtamalin) [inline README.md:22]: I would recommend moving this to be the first install option, with the caveat that it may not install the latest version.

Also suggest registering the relevant openSUSE or SLE repo from https://build.opensuse.org/project/show/Cloud:Tools to get access to the latest packaged version, though that may come with extra python dependencies.

File: README.md
Comment (rtamalin) [inline README.md:69]: Again pip installs should be done within an activated venv to avoid potential system Python infrastructure corruption.

File: README.md
Comment (rjschwei) [reply README.md:16]: agreed

File: README.md
Comment (rjschwei) [reply README.md:22]: agreed

File: README.md
Comment (rjschwei) [reply README.md:28]: yep

File: README.md
Comment (rjschwei) [inline README.md:62]: Echoing the help into the README means we have to touch the README every time there is a change to the cli. This will be forgotten and thus get things out of sync. We should not deliberately create more sources for bugs or potential misinformation.

File: README.md
Comment (rtamalin) [inline README.md:12]: In general looks good. However, I would recommend either explicitly showing the use of virtualenvs to hold what is pip installed, or else modify the pip command lines to explicitly show the use of the `--user` option on the `pip install` commands to indicate that it should install to the `~/.local` rather than system infrastructure, and similarly the `sudo` command prefix to run the zypper command...

File: README.md
Comment (rtamalin) [reply README.md:16]: Potentially also suggest that this is the recommended install mechanism for the tool on non-SUSE distros, e.g. Debian/Ubuntu or RHEL/CentOS.

File: README.md
Comment (rtamalin) [reply README.md:28]: Potentially suggest two install approaches here:

1. Install latest version normally, i.e. `pip install` without `-e`
2. Install latest sources in developer mode, i.e. `pip install` withe `-e`

The later can be used by a developer for adhoc testing of the development code without having to remember to run a `pip install`.

File: README.md
Comment (rtamalin) [reply README.md:69]: In the long run we should consider adding a `tox.ini` file to manage running the tests; it will manage the setup/maintenance of the testing virtualenv for us and simplify this to a simple `tox` command to run the tests.

File: README.md
Comment (jeremy-moffitt) [reply README.md:62]: Thats a good point and my fault for one of the acceptance criteria in the ticket...  Perhaps just reference the help command here

File: README.md
Comment (jeremy-moffitt) [reply README.md:83]: seems to be fixed in the latest version, thanks!

**Review by jeremy-moffitt — APPROVED** [review]
With the latest changes the dev and test workflows both work for me. LGTM (pending re-review from the other reviewers of course).

File: README.md
Comment (rjschwei) [inline README.md:21]: This probably warrants a comment about the distros and repositories. For example SLES is "SUSE Linux" and that command will fail in a generic install as the Public Cloud Module will be missing.

File: README.md
Comment (rjschwei) [inline README.md:29]: This is not correct. `python3` will be a symlink to a python 3 interpreter and that is highly distribution dependent, could be 3.8, 3.10 or something else. The given command works equally well if python3 is symlinked to python3.8

File: README.md
Comment (rjschwei) [inline README.md:33]: Depends on the shell being used, should probably be pointed out that this works for bash. venv creation also generates a activate.sh file for csh and derivatives.

**Comment by jeremy-moffitt** [comment]
@rjschwei when time permits, please re-review to see if Charu's changes address your feedback. Thank you!

File: README.md
Comment (rjschwei) [inline README.md:16]: s/OpenSUSE/openSUSE/

**Review by rjschwei — CHANGES_REQUESTED** [review]

**Review by rjschwei — APPROVED** [review]

**Pull request merged**
